react/src/base/inputs/SprkCheckbox/SprkCheckboxGroup/SprkCheckboxGroup.js

```diff
@@ -1,85 +1,139 @@
-      hasHelperText = true;
-      helperId = element.props.id || uniqueId('sprk-helper-text-');
-      return React.cloneElement(element, { id: helperId });
-    }
-    return element;
-  });
+  renderChildren() {
+    const { children } = this.props;
+    let errorContainerID;
+    let helperTextID;
+    let ariaDescribedByArray = [];
+    let fieldsetAriaDescribedBy;
 
-  // For each element, if it has grandChildren, add ariaDescribedBy to those
-  let key = 0;
-  let elementsToRender = elementsToProcess;
-  if (hasErrorContainer || hasHelperText) {
-    elementsToRender = elementsToProcess.map((element) => {
-      key += 1;
-      let grandChildren = null;
-      if (element.props.children) {
-        grandChildren = addPropsToMatchingComponents(
-          element.props.children,
-          ['SprkCheckboxItem'],
-          {
-            ariaDescribedBy: [helperId, errorId].join(' '),
-          },
-        );
+    /*
+     * Store references to errorContainerID, helperTextID,
+     * & fieldset ariaDescribedBy.
+     */
+    React.Children.forEach(children, (child) => {
+      if (child.type.name === SprkFieldset.name) {
+        fieldsetAriaDescribedBy = child.props.ariaDescribedBy;
       }
-
-      if (element.type.name === 'SprkCheckboxItem') {
-        return React.cloneElement(element, {
-          ariaDescribedBy: [helperId, errorId].join(' '),
-          key: `sprk-checkbox-item-${key}`,
-          children: grandChildren || element.children,
+      if (
+        child.type.name === SprkFieldError.name ||
+        child.type.name === SprkErrorContainer.name
+      ) {
+        errorContainerID = child.props.id;
+      }
+      if (child.type.name === SprkHelperText.name) {
+        helperTextID = child.props.id;
+      }
+      // If the child has it's own children
+      // map through them to get values.
+      if (child.props && child.props.children) {
+        React.Children.forEach(child.props.children, (grandchild) => {
+          if (grandchild.type) {
+            if (grandchild.type.name === SprkFieldset.name) {
+              fieldsetAriaDescribedBy = grandchild.props.ariaDescribedBy;
+            }
+            if (
+              grandchild.type.name === SprkFieldError.name ||
+              grandchild.type.name === SprkErrorContainer.name
+            ) {
+              errorContainerID = grandchild.props.id;
+            }
+            if (grandchild.type.name === SprkHelperText.name) {
+              helperTextID = grandchild.props.id;
+            }
+          }
         });
       }
-
-      return React.cloneElement(element, {
-        key: `sprk-checkbox-outer-item-${key}`,
-        children: grandChildren || element.children,
-      });
     });
+    /* Check to see if the errorContainerID
+     * and helperTextID are in the ariaDescribedByArray.
+     * If they aren't, add it to the array.
+     */
+    if (helperTextID || errorContainerID) {
+      if (fieldsetAriaDescribedBy) {
```

Comment:
is it possible to combine these if statement? It seems like there's only one if statement inside of an if statement (line 64+65)

Reply:
same for 68+69,  73+74

Reply:
64 & 65 can't be combined. There are cases where there isn't a fieldset ariaDescribedby being passed in initially, but other things inside that function need to run. Combining 68 & 69, 73 & 74 should work though!